R/api.R

```diff
@@ -72,24 +73,27 @@ metadata <- function() {
 
 #' @porcelain
 #'   POST /scenario/run => json(scenarioRunResponse)
+#'   state queue :: queue
 #'   body data :: json(scenarioRunRequest)
-scenario_run <- function(data) {
-  # Returning sample response for now
-  json_verbatim(read_local_json("sample_scenario_run_response.json"))
+scenario_run <- function(queue, data) {
+  data <- jsonlite::parse_json(data)
+  run_id <- queue$queue_model_run(
+    data$parameters,
+    model_version = data$modelVersion
+  )
+  json_verbatim(list(runId = run_id))
 }
 
 #' @porcelain
-#'   GET /scenario/status => json(scenarioStatus)
-#'   query run_id :: string
-scenario_status <-  function(run_id) {
-  # Returning sample response for now
-  json_verbatim(read_local_json("sample_scenario_status_response.json"))
+#'   GET /scenario/status/<run_id:string> => json(scenarioStatus)
+#'   state queue :: queue
+scenario_status <-  function(queue, run_id) {
+  json_verbatim(queue$get_run_status(run_id))
```

Comment:
```suggestion
  lapply(queue$get_run_status(run_id), jsonlite::unbox)
```

which was working before because json_verbatim was unboxing things. I'm a bit curious now about the intent of that function as I thought that was mostly concerned with splicing json...

Reply:
As discussed, sounds like there's a couple of competing `json_verbatim`s - this is the one I'm using here, which I initially marked as "probably temporary" because it was getting used to re-serialise json read in from the sample files - but then proved useful for other things... It auto unboxes and sets NULLs to null. 
https://github.com/jameel-institute/daedalus.api/blob/main/R/util.R#L25

Maybe the auto-unboxing is dangerous though, as I think that will render single-value arrays as scalars, which you won't always want..

So, I could either rename this to something more sensible (`json_auto_format`??) or could take the approach you suggest of always being explicit about formatting requirements in each serialise. What do you think?